apps/frontend/src/app/2_molecules/NetworkSwitch/NetworkSwitch.tsx

```diff
@@ -0,0 +1,21 @@
+import React, { FC, ReactElement } from 'react';
+
+import { ChainId } from '@sovryn/ethers-provider';
+
+import { useChainStore } from '../../../hooks/useChainStore';
+import { NetworkBanner } from '../NetworkBanner/NetworkBanner';
+
+type NetworkSwitchProps = {
+  components: { [chain in ChainId]?: ReactElement };
+};
+
+export const NetworkSwitch: FC<NetworkSwitchProps> = ({ components }) => {
+  const { currentChainId } = useChainStore();
+
+  if (components[currentChainId] !== undefined) {
```

Comment:
I think I'd do something like this (not tested just an example) in order to avoid rerenders or recalculations. Anyway just an improvement

`const componentToRender = React.useMemo(() => {
  return components[currentChainId] || <NetworkBanner requiredChainId={Object.keys(components)[0] || 'default'} />;
}, [currentChainId, components]);`

Reply:
Cool! Implemented! Thanks!